Java error with Keyword protection when using Pippin, Guard of the Citadel

**Describe the bug**
I targeted Mother of Runes with Pippin declaring protection from creatures and it caused a soft lock of the game, but didn't crash Forge.

This one surprised me, because I know that I've used Pippin before and this definitely *didn't* happen. 

`[AI Preferences] AI profile Cautious was chosen for the lobby player Melody.
[AI Preferences] AI profile Default was chosen for the lobby player Marcus.
[AI Preferences] AI profile Default was chosen for the lobby player Picon Oakenhelm.
Game-15 > java.lang.RuntimeException: Error in Keyword Protection from creatures for card Mother of Runes
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:128)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:86)
	at forge.game.card.Card.getKeywordForStaticAbility(Card.java:4865)
	at forge.game.card.Card.addChangedCardKeywords(Card.java:4842)
	at forge.game.ability.effects.ProtectEffect.resolve(ProtectEffect.java:149)
	at forge.game.ability.AbilityApiBased.resolve(AbilityApiBased.java:41)
	at forge.game.ability.AbilityUtils.resolveApiAbility(AbilityUtils.java:1454)
	at forge.game.ability.AbilityUtils.resolve(AbilityUtils.java:1422)
	at forge.game.zone.MagicStack.resolveStack(MagicStack.java:553)
	at forge.game.phase.PhaseHandler.startFirstTurn(PhaseHandler.java:1152)
	at forge.game.GameAction.startGame(GameAction.java:2098)
	at forge.game.Match.startGame(Match.java:90)
	at forge.gamemodes.match.HostedMatch$2.run(HostedMatch.java:255)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.RuntimeException: unknown protection keyword: Protection from creatures
	at forge.game.card.CardFactoryUtil.getProtectionValid(CardFactoryUtil.java:683)
	at forge.game.card.CardFactoryUtil.addReplacementEffect(CardFactoryUtil.java:2577)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:116)
	... 15 more`

**Desktop (please complete the following information):**
 - Version: forge-gui-desktop-1.6.60-SNAPSHOT-12-29

--- a/forge-game/src/main/java/forge/game/ability/effects/ProtectEffect.java
+++ b/forge-game/src/main/java/forge/game/ability/effects/ProtectEffect.java
@@ -5,9 +5,12 @@ import java.util.Arrays;
 import java.util.Iterator;
 import java.util.List;
 
+import org.apache.commons.lang3.StringUtils;
+
 import com.google.common.collect.Lists;
 
 import forge.GameCommand;
+import forge.card.CardType;
 import forge.card.MagicColor;
 import forge.game.Game;
 import forge.game.ability.AbilityUtils;
@@ -129,8 +132,12 @@ public class ProtectEffect extends SpellAbilityEffect {
         }
 
         List<String> gainsKWList = Lists.newArrayList();
-        for (String color : gains) {
-            gainsKWList.add(TextUtil.concatWithSpace("Protection from", color));
+        for (String type : gains) {
+            if (isChoice && sa.getParam("Choices").equals("CardType")) {
+                gainsKWList.add("Protection:" + type);
+            }  else {
+                gainsKWList.add(TextUtil.concatWithSpace("Protection from", type));
+            }
         }
 
         tgtCards.addAll(CardUtil.getRadiance(sa));
@@ -176,6 +183,8 @@ public class ProtectEffect extends SpellAbilityEffect {
             if (choices.contains("AnyColor")) {
                 gains.addAll(MagicColor.Constant.ONLY_COLORS);
                 choices = choices.replaceAll("AnyColor,?", "");
+            } else if (choices.contains("CardType")) {
+                choices = StringUtils.join(CardType.getAllCardTypes(), ",");
             }
             // Add any remaining choices
             if (choices.length() > 0) {

--- a/forge-gui/res/cardsfolder/m/magmatic_galleon.txt
+++ b/forge-gui/res/cardsfolder/m/magmatic_galleon.txt
@@ -4,7 +4,7 @@ Types:Artifact Vehicle
 PT:5/5
 T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDamage | TriggerDescription$ When CARDNAME enters the battlefield, it deals 5 damage to target creature an opponent controls.
 SVar:TrigDamage:DB$ DealDamage | ValidTgts$ Creature.OppCtrl | TgtPrompt$ Select target creature an opponent controls | NumDmg$ 5
-T:Mode$ ExcessDamageAll | ValidTarget$ Creature.OppCtrl | CombatDamage$ False | Execute$ TrigTreasure | TriggerDescription$ Whenever one or more creatures your opponents control are dealt excess noncombat damage, create a Treasure token.
+T:Mode$ ExcessDamageAll | ValidTarget$ Creature.OppCtrl | CombatDamage$ False | TriggerZones$ Battlefield | Execute$ TrigTreasure | TriggerDescription$ Whenever one or more creatures your opponents control are dealt excess noncombat damage, create a Treasure token.
 SVar:TrigTreasure:DB$ Token | TokenScript$ c_a_treasure_sac
 K:Crew:2
 Oracle:When Magmatic Galleon enters the battlefield, it deals 5 damage to target creature an opponent controls.\nWhenever one or more creatures your opponents control are dealt excess noncombat damage, create a Treasure token.\nCrew 2

--- a/forge-gui/res/cardsfolder/p/pippin_guard_of_the_citadel.txt
+++ b/forge-gui/res/cardsfolder/p/pippin_guard_of_the_citadel.txt
@@ -4,5 +4,5 @@ Types:Legendary Creature Halfling Soldier
 PT:2/2
 K:Vigilance
 K:Ward:1
-A:AB$ Protection | Cost$ T | ValidTgts$ Creature.YouCtrl+Other | TgtPrompt$ Select another target creature you control | Gains$ Choice | Choices$ artifacts,enchantments,creatures,battles,instants,sorceries,planeswalkers,lands | SpellDescription$ Another target creature you control gains protection from the card type of your choice until end of turn. (It can't be blocked, targeted, dealt damage, enchanted, or equipped by anything of that type.)
+A:AB$ Protection | Cost$ T | ValidTgts$ Creature.YouCtrl+Other | TgtPrompt$ Select another target creature you control | Gains$ Choice | Choices$ CardType | SpellDescription$ Another target creature you control gains protection from the card type of your choice until end of turn. (It can't be blocked, targeted, dealt damage, enchanted, or equipped by anything of that type.)
 Oracle:Vigilance, ward {1}\n{T}: Another target creature you control gains protection from the card type of your choice until end of turn. (It can't be blocked, targeted, dealt damage, enchanted, or equipped by anything of that type.)

